### ISSUE TITLE
[FEATURE] - Add Mod Category Grouping within the ModFolder

### ISSUE THREAD

**Issue by ScootersWorkbench** [FSGModding/FSG_Mod_Assistant#270]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

As a user I would like to see the Mod folder list of mods groups by their <Category> defined in the <Mod>.xml.  The categories should be able to collapse and expand.   This feature could also be able to be turned off if the user doesn't want that view.   The search & filter actions should still show the mods as they do today,  no need to show based on the category.   The Mod list sorting can stay the way that it currently is.

### Describe the solution you'd like

Grouped by Category:

(-) ModFolder
     (-) Small Tractors   (shown as expanded)
                 Mod_Small_Tractor01
                 Mod_Small_Tractor02
     (-) Medium Tractors
                 Mod_Med_Tractor01
                 etc...
     (+) Large Tractors  (shown as collapsed)
     (+) Harvesters

This would allow for more organization and it will tie like mods together and not have to hunt as much when you don't know the name or filename of the mod.

**Comment by jtsage** [comment]
yeah.  yeah.  6.x branch.  But I'll ping you in the next day or two, lets talk this one through - there are some hiccups to it.  But yes, a good idea, and something that will be possible in the 6.x release, as I can now parse the stuff *inside* the mod on the first pass, so long as I don't do the icons too - it gives a ton more info of what you are actually working with, which is way cool

**Comment by jtsage** [comment]
I've added the category, brand to the search options, I'll add it to the sort as well.
